routes/verify.go

```diff
@@ -452,3 +484,380 @@ func (fes *APIServer) sendEmail(email *mail.SGMailV3) {
+		return
+	}
+
+	// Create the request
+	req, err = http.NewRequest("POST", "https://netverify.com/api/v4/initiate", bytes.NewBuffer(jsonData))
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioBegin: Request creation failed: %v", err))
+		return
+	}
+
+	// Set content-type and basic authentication (token, secret)
+	req.Header.Set("Content-Type", "application/json")
+	req.SetBasicAuth(fes.Config.JumioToken, fes.Config.JumioSecret)
+
+	// Make the request
+	postRes, err := http.DefaultClient.Do(req)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioBegin: Request failed: %v", err))
+		return
+	}
+
+	if postRes.StatusCode != 200 {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioBegin: Request returned non-200 status code: %v", postRes.StatusCode))
+		return
+	}
+
+	// Decode the response
+	jumioInit := JumioInitResponse{}
+	if err = json.NewDecoder(postRes.Body).Decode(&jumioInit); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioBegin: Decode failed: %v", err))
+		return
+	}
+	if err = postRes.Body.Close(); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioBegin: Closing post request body failed: %v", err))
+		return
+	}
+
+	res := JumioBeginResponse{
+		URL: jumioInit.RedirectURL,
+	}
+	if err = json.NewEncoder(ww).Encode(res); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioBegin: Encode failed: %v", err))
+		return
+	}
+}
+
+type JumioFlowFinishedRequest struct {
+	PublicKey string
+	JumioInternalReference string
+	JWT string
+}
+
+func (fes *APIServer) JumioFlowFinished(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := JumioFlowFinishedRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioFlowFinished: Problem parsing request body: %v", err))
+		return
+	}
+
+	// Get UserMetadata from global state and check internal reference matches and we haven't marked this user as finished already.
+	userMetadata, err := fes.getUserMetadataFromGlobalState(requestData.PublicKey)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioFlowFinished: Problem getting user metadata from global state: %v", err))
+		return
+	}
+
+	if userMetadata.JumioInternalReference != requestData.JumioInternalReference {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioFlowFinished: UserMetadata's jumio internal reference (%v) does not match value from payload (%v)", userMetadata.JumioInternalReference, requestData.JumioInternalReference))
+		return
+	}
+
+	userMetadata.JumioFinishedTime = uint64(time.Now().UnixNano())
+
+	if err = fes.putUserMetadataInGlobalState(userMetadata); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioFlowFinished: Error putting jumio finish time in global state: %v", err))
+		return
+	}
+}
+
+// Jumio webhook - If Jumio verified user is a human that we haven't paid already, pay them some starter CLOUT.
```

Comment:
Needs a comment about blocking non authorized IPs. We should probably figure out the best place to document this for both Wyre and Jumio. Deep in the code isn't a great spot.